src/tools/crawler/network.rs

```diff
@@ -34,7 +39,9 @@ impl KnownNetwork {
         if !nodes.contains_key(&addr) {
             error!("set_connected(): addr not found ({})", addr);
         } else {
-            nodes.get_mut(&addr).unwrap().handshake_time = Some(handshake_time);
+            let mut node = nodes.get_mut(&addr).unwrap();
```

Comment:
why aren't we doing `if let Some(node) = nodes.get_mut(&addr)` as the first branch? we would avoid a double query that way

Reply:
Done